saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/authentication/Saml2ValidationError.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2002-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.saml2.provider.service.authentication;
+
+import org.springframework.security.core.SpringSecurityCoreVersion;
+import org.springframework.util.Assert;
+
+import java.io.Serializable;
+
+/**
+ * A representation of an SAML 2.0 Error.
+ *
+ * <p>
+ * At a minimum, an error response will contain an error code.
+ * The commonly used error code are defined in this class
+ * or a new codes can be defined in the future as arbitrary strings.
+ *
+ * @since 5.2
+ */
+public class Saml2ValidationError implements Serializable {
```

Comment:
I think it would be nice if the error and the codes were in different classes like they are with the OAuth 2.0 support. For example, the OAuth 2.0 support has `OAuth2Error` and `OAuth2ErrorCodes`.

Reply:
I'm also curious about the name. I like how this is patterned after `OAuth2Error` - is `Saml2Error` somehow wrong?

Reply:
The name `Saml2Error` makes sense after we've abstracted out the error codes into a separate class.